migrator/src/main/java/com/google/phonenumbers/migrator/MetadataZipFileReader.java

```diff
@@ -0,0 +1,103 @@
+import com.google.common.base.Preconditions;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * Represents a standard compliant metadata zip file where each {@link MetadataZipFileReader} zip
+ * file contains {@link CsvTable}'s that can be imported based on specified BCP47 version region
+ * codes without the need to extract the whole zip file.
+ */
+public final class MetadataZipFileReader {
+
+  private final ZipFile metadataZipFile;
+  private final String zipName;
+
+  private MetadataZipFileReader(String fileLocation, String zipName) throws IOException {
+    this.metadataZipFile = new ZipFile(fileLocation);
+    this.zipName = zipName;
+  }
+
+  /**
+   * Returns a MetadataZipFileReader for the given string (e.g. "./metadata.zip").
+   */
+  public static MetadataZipFileReader of(String fileLocation) throws IOException {
+    Preconditions.checkNotNull(fileLocation);
+    Preconditions.checkArgument(new File(fileLocation).exists(),
+        "Invalid zip file location: ", fileLocation);
+
+    String[] dirPaths = fileLocation.split("/");
+    String zipName = dirPaths[dirPaths.length - 1].isEmpty() ?
+        dirPaths[dirPaths.length - 2] : dirPaths[dirPaths.length - 1];
+    return new MetadataZipFileReader(fileLocation, zipName.split(".zip")[0]);
+  }
+
+  /**
+   * Returns the {@link CsvTable} for the given BCP47 region code (e.g. "44") if present.
+   */
+  public Optional<CsvTable<RangeKey>> importCsvTable(String regionCode) throws IOException {
+    String csvTableLocation = zipName + "/" + regionCode + "/ranges.csv";
+    ZipEntry csvFile = metadataZipFile.getEntry(csvTableLocation);
+
+    if (csvFile == null) {
+      return Optional.empty();
+    }
+    return Optional.of(CsvTable.importCsv(RangesTableSchema.SCHEMA,
+        new InputStreamReader(metadataZipFile.getInputStream(csvFile))));
+  }
+
+  public static void main(String[] args) throws Exception {
+    String fileLocation;
+    String regionCode;
+
+    if (args.length < 2) {
+      Scanner scanner = new Scanner(System.in);
+      System.out.println("You have not entered correct command line arguments");
+
+      System.out.print("\tPlease enter a zip file location of metadata csv files: ");
+      fileLocation = scanner.next();
+
+      System.out.print("\tPlease enter the region code of the ranges you want to import: ");
+      regionCode = scanner.next();
+    } else {
+      fileLocation = args[0];
+      regionCode = args[1];
+    }
+
+    MetadataZipFileReader m = MetadataZipFileReader.of(fileLocation);
+    Optional<CsvTable<RangeKey>> ranges = m.importCsvTable(regionCode);
+    ranges.orElseThrow(() -> new Exception("Region code not supported in zipfile"));
+
+    System.out.println("Table imported!");
+    ImmutableSet<RangeKey> columns = ranges.get().getKeys();
```

Comment:
[Tip] You can use "ranges.get().getKeys().forEach(System.out::println)" to do this in one line.